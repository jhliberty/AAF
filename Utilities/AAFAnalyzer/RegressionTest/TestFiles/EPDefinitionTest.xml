<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE aaf-file SYSTEM "EPDerivationChain.dtd">
<aaf-file>
 <dictionary>
  <operation-def description="Should cause REQ_EP_162 info result." name="Unused"/>
 </dictionary>
<!-- Test Operation Definitions -->
 <top-level name="Valid OperationGroup/OperationDefinition">
  <timeline-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
   <mono-audio-dissolve length="0" track-type="audio"/>
  </timeline-mob-slot>
 </top-level>
 <top-level name="Invalid OperationGroup/OperationDefinition">
  <timeline-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
   <operation-group length="0" operation-definition="chroma-key" track-type="audio"/>
  </timeline-mob-slot>
 </top-level>
<!-- Test every operation defined in the edit protocol with current Data
         Definition values -->
 <top-level name="Video Dissolve not in Transition">
  <timeline-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
   <sequence length="0" track-type="picture">
    <eoc track-type="picture" length="1"/>
    <video-dissolve track-type="picture" length="1"/>
    <eoc track-type="picture" length="1"/>
   </sequence>
  </timeline-mob-slot>
 </top-level>
 <top-level name="SMPTE Video Wipe not in Transition">
  <timeline-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
   <sequence length="0" track-type="picture">
    <eoc track-type="picture" length="1"/>
    <smpte-video-wipe track-type="picture" length="1"/>
    <eoc track-type="picture" length="1"/>
   </sequence>
  </timeline-mob-slot>
 </top-level>
 <top-level name="Video Speed Control not in Transition">
  <timeline-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
   <sequence length="0" track-type="picture">
    <eoc track-type="picture" length="1"/>
    <video-speed-control track-type="picture" length="1">
     <constant-speed-ratio denominator="1" numerator="1"/>
    </video-speed-control>
    <eoc track-type="picture" length="1"/>
   </sequence>
  </timeline-mob-slot>
 </top-level>
 <top-level name="Video Repeat not in Transition">
  <timeline-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
   <sequence length="0" track-type="picture">
    <eoc track-type="picture" length="1"/>
    <video-repeat track-type="picture" length="1"/>
    <eoc track-type="picture" length="1"/>
   </sequence>
  </timeline-mob-slot>
 </top-level>
 <top-level name="Video Flip not in Transition">
  <timeline-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
   <sequence length="0" track-type="picture">
    <eoc track-type="picture" length="1"/>
    <video-flip track-type="picture" length="1"/>
    <eoc track-type="picture" length="1"/>
   </sequence>
  </timeline-mob-slot>
 </top-level>
 <top-level name="Video Flop not in Transition">
  <timeline-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
   <sequence length="0" track-type="picture">
    <eoc track-type="picture" length="1"/>
    <video-flop track-type="picture" length="1"/>
    <eoc track-type="picture" length="1"/>
   </sequence>
  </timeline-mob-slot>
 </top-level>
 <top-level name="Video Flip-Flop not in Transition">
  <timeline-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
   <sequence length="0" track-type="picture">
    <eoc track-type="picture" length="1"/>
    <video-flip-flop track-type="picture" length="1"/>
    <eoc track-type="picture" length="1"/>
   </sequence>
  </timeline-mob-slot>
 </top-level>
 <top-level name="Video Position not in Transition">
  <timeline-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
   <sequence length="0" track-type="picture">
    <eoc track-type="picture" length="1"/>
    <video-position track-type="picture" length="1"/>
    <eoc track-type="picture" length="1"/>
   </sequence>
  </timeline-mob-slot>
 </top-level>
 <top-level name="Video Crop not in Transition">
  <timeline-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
   <sequence length="0" track-type="picture">
    <eoc track-type="picture" length="1"/>
    <video-crop track-type="picture" length="1"/>
    <eoc track-type="picture" length="1"/>
   </sequence>
  </timeline-mob-slot>
 </top-level>
 <top-level name="Video Scale not in Transition">
  <timeline-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
   <sequence length="0" track-type="picture">
    <eoc track-type="picture" length="1"/>
    <video-scale track-type="picture" length="1"/>
    <eoc track-type="picture" length="1"/>
   </sequence>
  </timeline-mob-slot>
 </top-level>
 <top-level name="Video Rotate not in Transition">
  <timeline-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
   <sequence length="0" track-type="picture">
    <eoc track-type="picture" length="1"/>
    <video-rotate track-type="picture" length="1"/>
    <eoc track-type="picture" length="1"/>
   </sequence>
  </timeline-mob-slot>
 </top-level>
 <top-level name="Video Corner Pinning not in Transition">
  <timeline-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
   <sequence length="0" track-type="picture">
    <eoc track-type="picture" length="1"/>
    <video-corner-pinning track-type="picture" length="1"/>
    <eoc track-type="picture" length="1"/>
   </sequence>
  </timeline-mob-slot>
 </top-level>
 <top-level name="Alpha With Video Key not in Transition">
  <timeline-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
   <sequence length="0" track-type="picture">
    <eoc track-type="picture" length="1"/>
    <alpha-with-video-key track-type="picture" length="1"/>
    <eoc track-type="picture" length="1"/>
   </sequence>
  </timeline-mob-slot>
 </top-level>
 <top-level name="Separate Alpha Key not in Transition">
  <timeline-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
   <sequence length="0" track-type="picture">
    <eoc track-type="picture" length="1"/>
    <separate-alpha-key track-type="picture" length="1"/>
    <eoc track-type="picture" length="1"/>
   </sequence>
  </timeline-mob-slot>
 </top-level>
 <top-level name="Luminance Key not in Transition">
  <timeline-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
   <sequence length="0" track-type="picture">
    <eoc track-type="picture" length="1"/>
    <luminance-key track-type="picture" length="1"/>
    <eoc track-type="picture" length="1"/>
   </sequence>
  </timeline-mob-slot>
 </top-level>
 <top-level name="Chroma Key not in Transition">
  <timeline-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
   <sequence length="0" track-type="picture">
    <eoc track-type="picture" length="1"/>
    <chroma-key track-type="picture" length="1"/>
    <eoc track-type="picture" length="1"/>
   </sequence>
  </timeline-mob-slot>
 </top-level>
 <top-level name="Mono Audio Gain not in Transition">
  <timeline-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
   <sequence length="0" track-type="audio">
    <eoc track-type="audio" length="1"/>
    <mono-audio-gain track-type="audio" length="1"/>
    <eoc track-type="audio" length="1"/>
   </sequence>
  </timeline-mob-slot>
 </top-level>
 <top-level name="Mono Audio Pan not in Transition">
  <timeline-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
   <sequence length="0" track-type="audio">
    <eoc track-type="audio" length="1"/>
    <mono-audio-pan track-type="audio" length="1"/>
    <eoc track-type="audio" length="1"/>
   </sequence>
  </timeline-mob-slot>
 </top-level>
 <top-level name="Mono Audio Dissolve not in Transition">
  <timeline-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
   <sequence length="0" track-type="audio">
    <eoc track-type="audio" length="1"/>
    <mono-audio-dissolve track-type="audio" length="1"/>
    <eoc track-type="audio" length="1"/>
   </sequence>
  </timeline-mob-slot>
 </top-level>
 <top-level name="Two Parameter Mobo Audio Dissolve not in Transition">
  <timeline-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
   <sequence length="0" track-type="audio">
    <eoc track-type="audio" length="1"/>
    <two-parameter-mono-audio-dissolve track-type="audio" length="1"/>
    <eoc track-type="audio" length="1"/>
   </sequence>
  </timeline-mob-slot>
 </top-level>
</aaf-file>
