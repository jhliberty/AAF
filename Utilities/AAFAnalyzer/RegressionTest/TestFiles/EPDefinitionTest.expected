        Name:   Edit Protocol Definition Test
        Desc:   Verify the use of DataDefinition objects throught the file.
        Result: Failed
        Reason: 
        Covered Requirements: None
        Noted Requirements: None
        Passing Requirements: REQ_EP_163
        Info Requirements: None
        Warning Requirements: None
        Failing Requirements: REQ_EP_162

            Name:   OperationDefinition Data Definition Value
            Desc:   An OperationDefinition object in an AAF file shall have a data definition value consistent with the data definition of an OperationGroup object that references it.
            Node:   8 (OperationGroup)
            Result: Failed
            Reason: OperationGroup in Mob slot with ID = 1 of Top-Level Composition "Invalid OperationGroup/OperationDefinition" has data definition value "DataDef_Sound" and references OperationDefinition "Chroma Key" with data definition value "DataDef_Picture".
            Covered Requirements: None
            Noted Requirements: None
            Passing Requirements: None
            Info Requirements: None
            Warning Requirements: None
            Failing Requirements: REQ_EP_162

            Name:   OperationDefinition Data Definition Value
            Desc:   An OperationDefinition object in an AAF file shall have a data definition value consistent with the data definition of an OperationGroup object that references it.
            Node:   131 (OperationDefinition)
            Result: Info
            Reason: OperationDefinition "Unused" is not referenced.
            Covered Requirements: None
            Noted Requirements: None
            Passing Requirements: None
            Info Requirements: REQ_EP_162
            Warning Requirements: None
            Failing Requirements: None

            Name:   Effect Data Definition Values
            Desc:   The effects defined in the Edit Protocol should be used with the AAF v1.1 data definition values of DataDef_Picture and DataDef_Sound.  The deprecated data definition values of DataDef_LegacyPicture and DataDef_LegacySound should only be used when working with legacy AAF v1.0 files.
            Node:   134 (OperationDefinition)
            Result: Passed
            Reason: OperationDefinition "Alpha With Video Key effect" has DataDefinition = DataDef_DataDef_Picture in an AAF Version 1.1 file.
            Covered Requirements: None
            Noted Requirements: None
            Passing Requirements: REQ_EP_163
            Info Requirements: None
            Warning Requirements: None
            Failing Requirements: None

            Name:   Effect Data Definition Values
            Desc:   The effects defined in the Edit Protocol should be used with the AAF v1.1 data definition values of DataDef_Picture and DataDef_Sound.  The deprecated data definition values of DataDef_LegacyPicture and DataDef_LegacySound should only be used when working with legacy AAF v1.0 files.
            Node:   151 (OperationDefinition)
            Result: Passed
            Reason: OperationDefinition "Chroma Key" has DataDefinition = DataDef_DataDef_Picture in an AAF Version 1.1 file.
            Covered Requirements: None
            Noted Requirements: None
            Passing Requirements: REQ_EP_163
            Info Requirements: None
            Warning Requirements: None
            Failing Requirements: None

            Name:   Effect Data Definition Values
            Desc:   The effects defined in the Edit Protocol should be used with the AAF v1.1 data definition values of DataDef_Picture and DataDef_Sound.  The deprecated data definition values of DataDef_LegacyPicture and DataDef_LegacySound should only be used when working with legacy AAF v1.0 files.
            Node:   132 (OperationDefinition)
            Result: Passed
            Reason: OperationDefinition "Luminance Key" has DataDefinition = DataDef_DataDef_Picture in an AAF Version 1.1 file.
            Covered Requirements: None
            Noted Requirements: None
            Passing Requirements: REQ_EP_163
            Info Requirements: None
            Warning Requirements: None
            Failing Requirements: None

            Name:   Effect Data Definition Values
            Desc:   The effects defined in the Edit Protocol should be used with the AAF v1.1 data definition values of DataDef_Picture and DataDef_Sound.  The deprecated data definition values of DataDef_LegacyPicture and DataDef_LegacySound should only be used when working with legacy AAF v1.0 files.
            Node:   142 (OperationDefinition)
            Result: Passed
            Reason: OperationDefinition "Mono Audio Dissolve" has DataDefinition = DataDef_DataDef_Sound in an AAF Version 1.1 file.
            Covered Requirements: None
            Noted Requirements: None
            Passing Requirements: REQ_EP_163
            Info Requirements: None
            Warning Requirements: None
            Failing Requirements: None

            Name:   Effect Data Definition Values
            Desc:   The effects defined in the Edit Protocol should be used with the AAF v1.1 data definition values of DataDef_Picture and DataDef_Sound.  The deprecated data definition values of DataDef_LegacyPicture and DataDef_LegacySound should only be used when working with legacy AAF v1.0 files.
            Node:   149 (OperationDefinition)
            Result: Passed
            Reason: OperationDefinition "Mono Audio Gain" has DataDefinition = DataDef_DataDef_Sound in an AAF Version 1.1 file.
            Covered Requirements: None
            Noted Requirements: None
            Passing Requirements: REQ_EP_163
            Info Requirements: None
            Warning Requirements: None
            Failing Requirements: None

            Name:   Effect Data Definition Values
            Desc:   The effects defined in the Edit Protocol should be used with the AAF v1.1 data definition values of DataDef_Picture and DataDef_Sound.  The deprecated data definition values of DataDef_LegacyPicture and DataDef_LegacySound should only be used when working with legacy AAF v1.0 files.
            Node:   148 (OperationDefinition)
            Result: Passed
            Reason: OperationDefinition "Mono Audio Pan" has DataDefinition = DataDef_DataDef_Sound in an AAF Version 1.1 file.
            Covered Requirements: None
            Noted Requirements: None
            Passing Requirements: REQ_EP_163
            Info Requirements: None
            Warning Requirements: None
            Failing Requirements: None

            Name:   Effect Data Definition Values
            Desc:   The effects defined in the Edit Protocol should be used with the AAF v1.1 data definition values of DataDef_Picture and DataDef_Sound.  The deprecated data definition values of DataDef_LegacyPicture and DataDef_LegacySound should only be used when working with legacy AAF v1.0 files.
            Node:   143 (OperationDefinition)
            Result: Passed
            Reason: OperationDefinition "SMPTE Video Wipe" has DataDefinition = DataDef_DataDef_Picture in an AAF Version 1.1 file.
            Covered Requirements: None
            Noted Requirements: None
            Passing Requirements: REQ_EP_163
            Info Requirements: None
            Warning Requirements: None
            Failing Requirements: None

            Name:   Effect Data Definition Values
            Desc:   The effects defined in the Edit Protocol should be used with the AAF v1.1 data definition values of DataDef_Picture and DataDef_Sound.  The deprecated data definition values of DataDef_LegacyPicture and DataDef_LegacySound should only be used when working with legacy AAF v1.0 files.
            Node:   135 (OperationDefinition)
            Result: Passed
            Reason: OperationDefinition "Separate Alpha Key effect" has DataDefinition = DataDef_DataDef_Picture in an AAF Version 1.1 file.
            Covered Requirements: None
            Noted Requirements: None
            Passing Requirements: REQ_EP_163
            Info Requirements: None
            Warning Requirements: None
            Failing Requirements: None

            Name:   Effect Data Definition Values
            Desc:   The effects defined in the Edit Protocol should be used with the AAF v1.1 data definition values of DataDef_Picture and DataDef_Sound.  The deprecated data definition values of DataDef_LegacyPicture and DataDef_LegacySound should only be used when working with legacy AAF v1.0 files.
            Node:   146 (OperationDefinition)
            Result: Passed
            Reason: OperationDefinition "Two-Parameter Mono Audio Dissolve" has DataDefinition = DataDef_DataDef_Sound in an AAF Version 1.1 file.
            Covered Requirements: None
            Noted Requirements: None
            Passing Requirements: REQ_EP_163
            Info Requirements: None
            Warning Requirements: None
            Failing Requirements: None

            Name:   Effect Data Definition Values
            Desc:   The effects defined in the Edit Protocol should be used with the AAF v1.1 data definition values of DataDef_Picture and DataDef_Sound.  The deprecated data definition values of DataDef_LegacyPicture and DataDef_LegacySound should only be used when working with legacy AAF v1.0 files.
            Node:   136 (OperationDefinition)
            Result: Passed
            Reason: OperationDefinition "Video Corner Pinning" has DataDefinition = DataDef_DataDef_Picture in an AAF Version 1.1 file.
            Covered Requirements: None
            Noted Requirements: None
            Passing Requirements: REQ_EP_163
            Info Requirements: None
            Warning Requirements: None
            Failing Requirements: None

            Name:   Effect Data Definition Values
            Desc:   The effects defined in the Edit Protocol should be used with the AAF v1.1 data definition values of DataDef_Picture and DataDef_Sound.  The deprecated data definition values of DataDef_LegacyPicture and DataDef_LegacySound should only be used when working with legacy AAF v1.0 files.
            Node:   144 (OperationDefinition)
            Result: Passed
            Reason: OperationDefinition "Video Crop" has DataDefinition = DataDef_DataDef_Picture in an AAF Version 1.1 file.
            Covered Requirements: None
            Noted Requirements: None
            Passing Requirements: REQ_EP_163
            Info Requirements: None
            Warning Requirements: None
            Failing Requirements: None

            Name:   Effect Data Definition Values
            Desc:   The effects defined in the Edit Protocol should be used with the AAF v1.1 data definition values of DataDef_Picture and DataDef_Sound.  The deprecated data definition values of DataDef_LegacyPicture and DataDef_LegacySound should only be used when working with legacy AAF v1.0 files.
            Node:   141 (OperationDefinition)
            Result: Passed
            Reason: OperationDefinition "Video Dissolve" has DataDefinition = DataDef_DataDef_Picture in an AAF Version 1.1 file.
            Covered Requirements: None
            Noted Requirements: None
            Passing Requirements: REQ_EP_163
            Info Requirements: None
            Warning Requirements: None
            Failing Requirements: None

            Name:   Effect Data Definition Values
            Desc:   The effects defined in the Edit Protocol should be used with the AAF v1.1 data definition values of DataDef_Picture and DataDef_Sound.  The deprecated data definition values of DataDef_LegacyPicture and DataDef_LegacySound should only be used when working with legacy AAF v1.0 files.
            Node:   139 (OperationDefinition)
            Result: Passed
            Reason: OperationDefinition "Video Flip Flop" has DataDefinition = DataDef_DataDef_Picture in an AAF Version 1.1 file.
            Covered Requirements: None
            Noted Requirements: None
            Passing Requirements: REQ_EP_163
            Info Requirements: None
            Warning Requirements: None
            Failing Requirements: None

            Name:   Effect Data Definition Values
            Desc:   The effects defined in the Edit Protocol should be used with the AAF v1.1 data definition values of DataDef_Picture and DataDef_Sound.  The deprecated data definition values of DataDef_LegacyPicture and DataDef_LegacySound should only be used when working with legacy AAF v1.0 files.
            Node:   138 (OperationDefinition)
            Result: Passed
            Reason: OperationDefinition "Video Flip" has DataDefinition = DataDef_DataDef_Picture in an AAF Version 1.1 file.
            Covered Requirements: None
            Noted Requirements: None
            Passing Requirements: REQ_EP_163
            Info Requirements: None
            Warning Requirements: None
            Failing Requirements: None

            Name:   Effect Data Definition Values
            Desc:   The effects defined in the Edit Protocol should be used with the AAF v1.1 data definition values of DataDef_Picture and DataDef_Sound.  The deprecated data definition values of DataDef_LegacyPicture and DataDef_LegacySound should only be used when working with legacy AAF v1.0 files.
            Node:   140 (OperationDefinition)
            Result: Passed
            Reason: OperationDefinition "Video Flop" has DataDefinition = DataDef_DataDef_Picture in an AAF Version 1.1 file.
            Covered Requirements: None
            Noted Requirements: None
            Passing Requirements: REQ_EP_163
            Info Requirements: None
            Warning Requirements: None
            Failing Requirements: None

            Name:   Effect Data Definition Values
            Desc:   The effects defined in the Edit Protocol should be used with the AAF v1.1 data definition values of DataDef_Picture and DataDef_Sound.  The deprecated data definition values of DataDef_LegacyPicture and DataDef_LegacySound should only be used when working with legacy AAF v1.0 files.
            Node:   137 (OperationDefinition)
            Result: Passed
            Reason: OperationDefinition "Video Position" has DataDefinition = DataDef_DataDef_Picture in an AAF Version 1.1 file.
            Covered Requirements: None
            Noted Requirements: None
            Passing Requirements: REQ_EP_163
            Info Requirements: None
            Warning Requirements: None
            Failing Requirements: None

            Name:   Effect Data Definition Values
            Desc:   The effects defined in the Edit Protocol should be used with the AAF v1.1 data definition values of DataDef_Picture and DataDef_Sound.  The deprecated data definition values of DataDef_LegacyPicture and DataDef_LegacySound should only be used when working with legacy AAF v1.0 files.
            Node:   147 (OperationDefinition)
            Result: Passed
            Reason: OperationDefinition "Video Repeat" has DataDefinition = DataDef_DataDef_Picture in an AAF Version 1.1 file.
            Covered Requirements: None
            Noted Requirements: None
            Passing Requirements: REQ_EP_163
            Info Requirements: None
            Warning Requirements: None
            Failing Requirements: None

            Name:   Effect Data Definition Values
            Desc:   The effects defined in the Edit Protocol should be used with the AAF v1.1 data definition values of DataDef_Picture and DataDef_Sound.  The deprecated data definition values of DataDef_LegacyPicture and DataDef_LegacySound should only be used when working with legacy AAF v1.0 files.
            Node:   133 (OperationDefinition)
            Result: Passed
            Reason: OperationDefinition "Video Rotate" has DataDefinition = DataDef_DataDef_Picture in an AAF Version 1.1 file.
            Covered Requirements: None
            Noted Requirements: None
            Passing Requirements: REQ_EP_163
            Info Requirements: None
            Warning Requirements: None
            Failing Requirements: None

            Name:   Effect Data Definition Values
            Desc:   The effects defined in the Edit Protocol should be used with the AAF v1.1 data definition values of DataDef_Picture and DataDef_Sound.  The deprecated data definition values of DataDef_LegacyPicture and DataDef_LegacySound should only be used when working with legacy AAF v1.0 files.
            Node:   150 (OperationDefinition)
            Result: Passed
            Reason: OperationDefinition "Video Scale" has DataDefinition = DataDef_DataDef_Picture in an AAF Version 1.1 file.
            Covered Requirements: None
            Noted Requirements: None
            Passing Requirements: REQ_EP_163
            Info Requirements: None
            Warning Requirements: None
            Failing Requirements: None

            Name:   Effect Data Definition Values
            Desc:   The effects defined in the Edit Protocol should be used with the AAF v1.1 data definition values of DataDef_Picture and DataDef_Sound.  The deprecated data definition values of DataDef_LegacyPicture and DataDef_LegacySound should only be used when working with legacy AAF v1.0 files.
            Node:   145 (OperationDefinition)
            Result: Passed
            Reason: OperationDefinition "Video Speed Control" has DataDefinition = DataDef_DataDef_Picture in an AAF Version 1.1 file.
            Covered Requirements: None
            Noted Requirements: None
            Passing Requirements: REQ_EP_163
            Info Requirements: None
            Warning Requirements: None
            Failing Requirements: None

